torchaudio/prototype/ctc_decoder/__init__.py

```diff
@@ -1,42 +1,29 @@
-_INITIALIZED = False
-_LAZILY_IMPORTED = [
-    "Hypothesis",
-    "CTCDecoder",
-    "ctc_decoder",
-    "lexicon_decoder",
-    "download_pretrained_files",
-]
+def __getattr__(name: str):
+    if name in ["ctc_decoder", "lexicon_decoder"]:
+        import warnings
 
+        from torchaudio.models.decoder import ctc_decoder
 
-def _init_extension():
-    import torchaudio
+        warnings.warn(
+            f"{__name__}.{name} has been moved to torchaudio.models.decoder.ctc_decoder",
+            DeprecationWarning,
+        )
 
-    torchaudio._extension._load_lib("libtorchaudio_decoder")
+        if name == "lexicon_decoder":
+            global lexicon_decoder
+            lexicon_decoder = ctc_decoder
+            return lexicon_decoder
+        else:
+            return ctc_decoder
+    elif name == "download_pretrained_files":
+        import warnings
 
-    global _INITIALIZED
-    _INITIALIZED = True
+        from torchaudio.models.decoder import download_pretrained_files
 
+        return download_pretrained_files
 
-def __getattr__(name: str):
```

Comment:
@mthrok would we need to keep this code still?

Reply:
Not 100% which part you are pointing but if you are referring to the lazy import mechanism, then yes.
Since we employ lazy initialization mechanism for CTC decoder in `torchaudio.models.decoder` module, we also need to do the same in `torchaudio.prototype.ctc_decoder`, otherwise the lazy initialization mechanism in `torchaudio.models.decoder` will be materialized as soon.

However, now that I think about it, the code and build for CTC decoder is 100% under our control (unlike the availability of CUDA or FFmpeg and such), so maybe we can get rid of the lazy initialization for CTC decoder. Let's follow up on it later. I think this PR is good to go.